base/reflection.jl

```diff
@@ -1463,19 +1463,31 @@ struct CodegenParams
     """
     lookup::Ptr{Cvoid}
 
+    """
+    A pointer to name of cpu target
+    """
+    targettriple::Ptr{UInt8}
+
+    """
+    A pointer to name of cpu type.
+    """
+    cpu::Ptr{UInt8}
+
     function CodegenParams(; track_allocations::Bool=true, code_coverage::Bool=true,
                    prefer_specsig::Bool=false,
                    gnu_pubnames::Bool=true, debug_info_kind::Cint = default_debug_info_kind(),
                    debug_info_level::Cint = Cint(JLOptions().debug_level), safepoint_on_entry::Bool=true,
                    gcstack_arg::Bool=true, use_jlplt::Bool=true,
-                   lookup::Ptr{Cvoid}=unsafe_load(cglobal(:jl_rettype_inferred_addr, Ptr{Cvoid})))
+                   lookup::Ptr{Cvoid}=unsafe_load(cglobal(:jl_rettype_inferred_addr, Ptr{Cvoid})),
+                   targettriple::Ptr{UInt8} = Sys.MACHINE |> pointer,
+                   cpu::Ptr{UInt8} = Sys.CPU_NAME |> pointer)
         return new(
             Cint(track_allocations), Cint(code_coverage),
             Cint(prefer_specsig),
             Cint(gnu_pubnames), debug_info_kind,
             debug_info_level, Cint(safepoint_on_entry),
             Cint(gcstack_arg), Cint(use_jlplt),
-            lookup)
+            lookup, targettriple, cpu)
```

Comment:
This doesn't work since `targettriple` or `cpu` are likely stemming from GC tracked strings and by turning them into pointers the GC loses track and might free the data associated with the pointer.